tests/test-amp-crowdsignal-embed-handler.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Class AMP_Crowdsignal_Embed_Test
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Crowdsignal_Embed_Test
+ *
+ * @covers AMP_Crowdsignal_Embed_Handler
+ */
+class AMP_Crowdsignal_Embed_Test extends WP_UnitTestCase {
+
+	/**
+	 * Get conversion data.
+	 *
+	 * @return array
+	 */
+	public function get_conversion_data() {
+		$poll_response   = array(
+			'type'          => 'rich',
+			'version'       => '1.0',
+			'provider_name' => 'Crowdsignal',
+			'provider_url'  => 'https://crowdsignal.com',
+			'title'         => 'Which design do you prefer?',
+			'html' => '<script type="text/javascript" charset="utf-8" src="https://secure.polldaddy.com/p/7012505.js"></script><noscript><a href="https://poll.fm/7012505">Which design do you prefer?</a></noscript>', // phpcs:ignore
+		);
+		$survey_response = array(
+			'type'          => 'rich',
+			'version'       => '1.0',
+			'provider_name' => 'Crowdsignal',
+			'provider_url'  => 'https://crowdsignal.com',
+			'html'          => '<div class="pd-embed" data-settings="{&quot;type&quot;:&quot;iframe&quot;,&quot;auto&quot;:true,&quot;domain&quot;:&quot;rydk.survey.fm&quot;,&quot;id&quot;:&quot;test-survey&quot;}"></div><script type="text/javascript">(function(d,c,j){if(!document.getElementById(j)){var pd=d.createElement(c),s;pd.id=j;pd.src=(\'https:\'==document.location.protocol)?\'https://polldaddy.com/survey.js\':\'http://i0.poll.fm/survey.js\';s=document.getElementsByTagName(c)[0];s.parentNode.insertBefore(pd,s);}}(document,\'script\',\'pd-embed\'));</script>',
+		);
+
+		return array(
+			'poll.fm'          => array(
+				'https://poll.fm/7012505',
+				'<p><a href="https://poll.fm/7012505">Which design do you prefer?</a></p>',
+				$poll_response,
+			),
+
+			'polldaddy_poll'   => array(
+				'https://polldaddy.com/poll/7012505/',
+				'<p><a href="https://poll.fm/7012505">Which design do you prefer?</a></p>',
+				$poll_response,
+			),
+
+			/*
+			 * There seems to be a bug in WordPress core's oEmbed handling. The following patch seems necessary to
+			 * wp-includes/class-oembed.php:
+			 *  - '#https?://survey\.fm/.*#i'                    => array( 'https://api.crowdsignal.com/oembed', true ),
+			 *  + '#https?://(\w+\.)?survey\.fm/.*#i'            => array( 'https://api.crowdsignal.com/oembed', true ),
+			 */
+			'survey.fm'        => array(
```

Comment:
This test is causing the build failure. @kraftbj See my comment above as there seems to be a problem with Crowdsignal's support in core.

Reply:
https://core.trac.wordpress.org/ticket/46467